R/join_ald_scenario.R

```diff
@@ -42,14 +42,27 @@ join_ald_scenario <- function(data,
   region_isos <- region_isos %>%
     dplyr::rename(scenario_source = .data$source)
 
-  data %>%
+  out <- data %>%
     left_join(ald, by = ald_columns()) %>%
```

Comment:
@jdhoffa, do you confirm the type of join we want here is `left_join()`? 

I see we later use `inner_join()`. 

Compare:

``` r
library(dplyr, warn.conflicts = FALSE)

data <- tibble(name_ald = "a", sector = "x")
ald <-  tibble(name_ald = "b", sector = "y")

left_join(data, ald)
#> Joining, by = c("name_ald", "sector")
#> # A tibble: 1 x 2
#>   name_ald sector
#>   <chr>    <chr> 
#> 1 a        x

inner_join(data, ald)
#> Joining, by = c("name_ald", "sector")
#> # A tibble: 0 x 2
#> # … with 2 variables: name_ald <chr>, sector <chr>
```

<sup>Created on 2020-05-28 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
Yes, I confirm that this is correct. 
In the first case, we are linking the portfolio to any matches in the ald, hence we want to keep all matches in the loanbook (this is useful later for coverage calculations). 

In the second case, we are trying to filter the first dataframe based on the second, and we want only rows that are common to both

Reply:
Thanks for checking